bin/ci/ubuntu/build-and-test.sh

```diff
@@ -10,26 +10,41 @@ echo "using TARGET: $TARGET"
 
 # Ensure APP defaults to rippled if it's not set.
 : ${APP:=rippled}
+echo "using APP: $APP"
 
 JOBS=${NUM_PROCESSORS:-2}
 if [[ ${TARGET} == *.nounity ]]; then
-    JOBS=$((2*${JOBS}))
+  JOBS=$((2*${JOBS}))
 fi
 
 if [[ ${BUILD:-scons} == "cmake" ]]; then
+  CMAKE_EXTRA_ARGS=""
   echo "cmake building ${APP}"
   CMAKE_TARGET=$CC.$TARGET
   if [[ ${CI:-} == true ]]; then
     CMAKE_TARGET=$CMAKE_TARGET.ci
   fi
+  if [[ ${USE_CCACHE:-} == true ]]; then
+    echo "using ccache with basedir [${CCACHE_BASEDIR:-}]"
+    CMAKE_EXTRA_ARGS+=" -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
+  fi
+  if [ -d "build/${CMAKE_TARGET}" ]; then
+    rm -rf "build/${CMAKE_TARGET}"
+  fi
   mkdir -p "build/${CMAKE_TARGET}"
   pushd "build/${CMAKE_TARGET}"
-  cmake ../.. -Dtarget=$CMAKE_TARGET
+  cmake ../.. -Dtarget=$CMAKE_TARGET ${CMAKE_EXTRA_ARGS}
   cmake --build . -- -j${JOBS}
+  if [[ ${BUILD_BOTH:-} == true ]]; then
+    if [[ ${TARGET} == *.unity ]]; then
+      cmake --build . --target rippled_classic -- -j$((2*${JOBS}))
```

Comment:
...for the same reason we did that it on line 17. I personally think this a bunch of complexity - I think we should pick one value for all builds and not muck with it in the script. Also, you'll probably point out by my logic here that line 25 should be `0.5*${JOBS}`, but I just couldn't bring myself to write that. I'm open to the suggestion of getting rid of $JOBS manipulation completely and going with one sane value.  If I had to pick, I would try to do what RPM builder does for `smp_flags`:  http://www.g-loaded.eu/2009/04/24/manually-prepare-the-rpm-building-environment/